src/lib.rs

```diff
@@ -73,17 +73,18 @@ pub use signature::SignedRequest;
 pub use request::default_tls_client;
 pub use region::default_region;
 
-mod param;
-mod region;
-mod request;
-mod xmlerror;
-mod xmlutil;
-mod serialization;
+pub mod param;
```

Comment:
Making everything public here is one of the uglier parts of this.  Exporting the mock library, for example, feels like it could be misleading.